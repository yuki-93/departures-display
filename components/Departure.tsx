import React from "react";

export interface DeparturesType {
    cancelled: boolean;
    direction: string;
    tripId: string;
    line: string;
    when: string | null;
    plannedWhen: string;
    delay: number | null;
}

export const Departure = ({ cancelled, direction, line, when, plannedWhen, delay }: DeparturesType): JSX.Element => {
    const whenDate = when !== null ? new Date(when) : null;
    const plannedWhenDate = new Date(plannedWhen);
    return (
        <div style={{border: "1px solid gray", borderRadius: "10px", padding: "10px", margin: "5px", flex: "0 1 300px", backgroundColor: "lightgreen"}}>
            <div>{line} - {direction}</div>
            <div style={{color: "red"}}>{`${cancelled ? "Fahrt fällt aus" : ""}`}</div>
            <div>{whenDate !== null && `aktuelle Uhrzeit: ${("0" + whenDate.getHours()).slice(-2)}:${("0" + whenDate.getMinutes()).slice(-2)}:${("0" + whenDate.getSeconds()).slice(-2)}`}</div>
            <div style={{color: "red"}}>{delay !== null && `Verspätung: ${delay/60} min`}</div>
            <div>{`geplante Uhrzeit: ${("0" + plannedWhenDate.getHours()).slice(-2)}:${("0" + plannedWhenDate.getMinutes()).slice(-2)}:${("0" + plannedWhenDate.getSeconds()).slice(-2)}`}</div>
        </div>  
    )
};