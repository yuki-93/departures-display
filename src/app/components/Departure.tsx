import React from "react";
import styles from "./Departure.module.scss";
import { DeparturesType } from "../../../types";

export const Departure = ({
  cancelled,
  direction,
  line,
  when,
  plannedWhen,
  delay,
}: DeparturesType): React.ReactNode => {
  const whenDate = when !== null ? new Date(when) : null;
  const plannedWhenDate = new Date(plannedWhen);

  return (
    <div
      className={`${styles["card-gray"]} ${
        delay !== null && delay < 0 ? styles["earlier"] : ""
      } ${delay !== null && delay > 0 ? styles["delayed"] : ""}`}
    >
      <div>
        {line} | {direction}
      </div>
      {delay !== null && (
        <>
          <div className={styles.cancelled}>{`${
            cancelled ? "Fahrt fällt aus" : ""
          }`}</div>
          <div
            className={
              styles[`card-title-${delay < 0 ? "earlier" : "delayed"}`]
            }
          >
            {delay !== 0 && `Verspätung: ${delay / 60} min`}
          </div>
        </>
      )}
      <div>
        {whenDate !== null &&
          `aktuelle Uhrzeit: ${("0" + whenDate.getHours()).slice(-2)}:${(
            "0" + whenDate.getMinutes()
          ).slice(-2)} Uhr`}
      </div>
      <div>{`geplante Uhrzeit: ${("0" + plannedWhenDate.getHours()).slice(
        -2
      )}:${("0" + plannedWhenDate.getMinutes()).slice(-2)} Uhr`}</div>
    </div>
  );
};
