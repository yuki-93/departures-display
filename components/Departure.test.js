import { render } from "@testing-library/react";
import { Departure } from "./Departure";

describe("Departure", () => {
    test("with nullable values", () => {
        const input = {
            when: null,
            delay: null,
            cancelled: true,
            direction: "Mocked Terminus",
            line: "42",
            tripId: "1234567890",
            plannedWhen: new Date(2020, 1, 1).getTime()
        }

        const wrapper = render(<Departure {...input} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("delayed 2min", () => {
        const input = {
            when: new Date(2020, 1, 1).getTime(),
            delay: 120,
            cancelled: false,
            direction: "Mocked Terminus",
            line: "42",
            tripId: "1234567890",
            plannedWhen: new Date(2020, 1, 1).getTime()
        }

        const wrapper = render(<Departure {...input} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("earlier 2min", () => {
        const input = {
            when: new Date(2020, 1, 1).getTime(),
            delay: -120,
            cancelled: false,
            direction: "Mocked Terminus",
            line: "42",
            tripId: "1234567890",
            plannedWhen: new Date(2020, 1, 1).getTime()
        }

        const wrapper = render(<Departure {...input} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("on time", () => {
        const input = {
            when: new Date(2020, 1, 1).getTime(),
            delay: null,
            cancelled: false,
            direction: "Mocked Terminus",
            line: "42",
            tripId: "1234567890",
            plannedWhen: new Date(2020, 1, 1).getTime()
        }

        const wrapper = render(<Departure {...input} />);

        expect(wrapper).toMatchSnapshot();
    });
})