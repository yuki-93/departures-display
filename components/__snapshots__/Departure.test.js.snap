// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Departure delayed 2min 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: rgba(255, 0, 0, 0.3);"
      >
        <div>
          42
           | 
          Mocked Terminus
        </div>
        <div
          style="color: red; font-weight: bold;"
        />
        <div
          style="color: red; font-weight: bold;"
        >
          Versp채tung: 2 min
        </div>
        <div>
          aktuelle Uhrzeit: 00:00 Uhr
        </div>
        <div>
          geplante Uhrzeit: 00:00 Uhr
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: rgba(255, 0, 0, 0.3);"
    >
      <div>
        42
         | 
        Mocked Terminus
      </div>
      <div
        style="color: red; font-weight: bold;"
      />
      <div
        style="color: red; font-weight: bold;"
      >
        Versp채tung: 2 min
      </div>
      <div>
        aktuelle Uhrzeit: 00:00 Uhr
      </div>
      <div>
        geplante Uhrzeit: 00:00 Uhr
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Departure earlier 3min 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: rgba(0, 255, 0, 0.3);"
      >
        <div>
          42
           | 
          Mocked Terminus
        </div>
        <div
          style="color: red; font-weight: bold;"
        />
        <div
          style="color: green; font-weight: bold;"
        >
          Versp채tung: -3 min
        </div>
        <div>
          aktuelle Uhrzeit: 00:00 Uhr
        </div>
        <div>
          geplante Uhrzeit: 00:00 Uhr
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: rgba(0, 255, 0, 0.3);"
    >
      <div>
        42
         | 
        Mocked Terminus
      </div>
      <div
        style="color: red; font-weight: bold;"
      />
      <div
        style="color: green; font-weight: bold;"
      >
        Versp채tung: -3 min
      </div>
      <div>
        aktuelle Uhrzeit: 00:00 Uhr
      </div>
      <div>
        geplante Uhrzeit: 00:00 Uhr
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Departure on time 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: lightblue;"
      >
        <div>
          42
           | 
          Mocked Terminus
        </div>
        <div>
          aktuelle Uhrzeit: 00:00 Uhr
        </div>
        <div>
          geplante Uhrzeit: 00:00 Uhr
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: lightblue;"
    >
      <div>
        42
         | 
        Mocked Terminus
      </div>
      <div>
        aktuelle Uhrzeit: 00:00 Uhr
      </div>
      <div>
        geplante Uhrzeit: 00:00 Uhr
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Departure with nullable values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: lightblue;"
      >
        <div>
          42
           | 
          Mocked Terminus
        </div>
        <div />
        <div>
          geplante Uhrzeit: 00:00 Uhr
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="border: 1px solid gray; border-radius: 10px; padding: 10px; margin: 10px; flex: 0 1 300px; background-color: lightblue;"
    >
      <div>
        42
         | 
        Mocked Terminus
      </div>
      <div />
      <div>
        geplante Uhrzeit: 00:00 Uhr
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
